Beta_Integrations/ProofpointThreatResponse/ProofpointThreatResponse.py

```diff
@@ -193,35 +192,461 @@ def delete_indicator_command():
+                         f'URL: {fullurl}, StatusCode: {incidents_list.status_code}')
+        else:
+            return_error(f'The operation failed. URL: {fullurl}, StatusCode: {incidents_list.status_code}')
+    return incidents_list.json()
+
+
+def get_incidents_batch_by_time_request(params):
+    """Perform an API request to get incidents from ProofPoint in batches to prevent a timeout.
+
+    Args:
+        params(dict): The params of the request
+
+    Returns:
+        list. The incidents returned from the API call
+    """
+    incidents_list = []  # type:list
+    time_delta = timedelta(days=1)  # batch by days
+
+    current_time = datetime.now()
+    created_after = datetime.strptime(params.get('created_after'), '%Y-%m-%dT%H:%M:%SZ')
+    created_before = created_after + time_delta
+    request_params = {
+        'state': params.get('state'),
+        'created_after': created_after.isoformat().split('.')[0] + 'Z',
+        'created_before': created_before.isoformat().split('.')[0] + 'Z'
+    }
+
+    while created_before < current_time:
+        incidents_list.extend(get_incidents_request(request_params))
+
+        # advancing fetch time one day forward
+        created_after = created_before
+        created_before = created_before + time_delta
+
+        # updating params according to the new times
+        request_params['created_after'] = created_after.isoformat().split('.')[0] + 'Z'
+        request_params['created_before'] = created_before.isoformat().split('.')[0] + 'Z'
+
+    # fetching the last batch
+    request_params['created_before'] = current_time.isoformat().split('.')[0] + 'Z'
+    incidents_list.extend(get_incidents_request(request_params))
+
+    return incidents_list
+
+
+def fetch_incidents_command():
+    """
+        Fetches incidents from the ProofPoint API.
+    """
+    integration_params = demisto.params()
+    last_fetch = demisto.getLastRun().get('last_fetch', {})
+    incidents_states = integration_params.get('states')
+    for state in incidents_states:
+        if not last_fetch.get(state):
+            last_fetch[state] = FETCH_TIME
+
+    incidents = []
+
+    for state in incidents_states:
+        request_params = {
+            'created_after': last_fetch[state],
+            'state': state
+        }
+
+        state_parsed_fetch = dateutil.parser.parse(last_fetch[state])
+        incidents_list = get_incidents_batch_by_time_request(request_params)
+        filtered_incidents_list = filter_incidents(incidents_list)
+        for incident in filtered_incidents_list:
+            incident_creation_time = dateutil.parser.parse(incident['created_at'])
+            if incident_creation_time > state_parsed_fetch:
+                id = incident.get('id')
+                inc = {
+                    'name': f'ProofPoint_TRAP - ID {id}',
+                    'rawJSON': json.dumps(incident),
+                    'occurred': incident['created_at']
```

Comment:
is it in a format that demisto can work with? because in the upper function you preform manipulations